RadioController.py

```diff
@@ -0,0 +1,112 @@
+from typing import Dict, Any, Callable, Union, List
+
+import RocketData
+import SubpacketIDs
+from SubpacketIDs import SubpacketEnum
+
+# Map subpacket id (int) to length (int) in bytes. Only includes types with CONSTANT lengths
+PACKET_ID_TO_CONST_LENGTH: Dict[int, int] = {
+    0x00: 4,
+    0x02: 2,
+    0x04: 25,
+    0x05: 0000,  # TODO ack length?
+    0x30: 42,
+}
+for i in SubpacketIDs.get_list_of_sensor_IDs():
+    PACKET_ID_TO_CONST_LENGTH[i] = 5
+
+
+# Check if packet of given type has constant length
+def isPacketLengthConst(subpacket_id):
+    return subpacket_id in PACKET_ID_TO_CONST_LENGTH.keys()
+
+
+# Return dict of subpacket id, length of subpacket in byte_list, subpacket typename, data_unit
+# Current implementation strips off subpacket id + length (if included), returning dictionary of id and data length
+def extract_subpacket(byte_list: List):
+    subpacket_id: int = int.from_bytes(byte_list[0], "big")
+    length: int = 0
+    # check that id is valid:
+    if not SubpacketIDs.isSubpacketID(subpacket_id):
+        raise ValueError
+    if isPacketLengthConst(subpacket_id):
+        length = PACKET_ID_TO_CONST_LENGTH[int.from_bytes(byte_list[0], "big")]
+        data_unit = byte_list[1:length]
+        del byte_list[0:1]
+        length = length - 1  # TODO Fix this bad code somehow
+    else:
+        length = int.from_bytes(byte_list[1], "big")
+        data_unit = byte_list[2:length]
+        del byte_list[0:2]
+        length = length - 2  # TODO Fix this bad code somehow
+    parsed_subpacket: Dict[str, int] = {'id': subpacket_id, 'length': length}
+    return parsed_subpacket
```

Comment:
It looks like this could just be:
```suggestion
    return subpacket_id, length
```

and then calling it would just be:
`subpacket_id, length = extract_subpacket(byte_list)`

Seems simpler and python always does stuff like this in its libraries I think. Although what I like about the dict is that you dont have to worry about ordering (but you do have to worry about strings ig).

Either way works!